### ISSUE TITLE
Installation fails on latest versions of Debian and Ubuntu

### ISSUE THREAD

**Issue by benmullins** [gboudreau/Greyhole#137]
On the latest releases of Debian and Ubuntu (Debian 8 'jessie', and Ubuntu 15.04) the apt-get installation of greyhole fails to complete correctly. I believe that this is because with Debian's switch to systemd, they removed the samba service wrapper that greyhole depends on, and now there are separate services for smbd and nmbd. Ubuntu seems to have inherited this change from upstream.

When I first install it (using the command given on greyhole.net) it happily installs all of the dependencies, and when it finally gets to the end this happens:

```
Setting up greyhole (0.9.61-1) ...
insserv: Service samba has to be enabled to start service greyhole
insserv: exiting now!
update-rc.d: error: insserv rejected the script header
dpkg: error processing package greyhole (--configure):
 subprocess installed post-installation script returned error exit status 1
Processing triggers for libc-bin (2.19-18) ...
Processing triggers for systemd (215-17+deb8u1) ...
Errors were encountered while processing:
 greyhole
E: Sub-processing /usr/bin/dpkg returned an error code (1)
root@debian:~# 
```

Trying to rerun ```apt-get install greyhole``` does more or less the same thing:

```
root@debian:~# apt-get install greyhole
Reading package lists... Done
Building dependency tree
Reading state information... Done
greyhole is already the newest version.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
1 not fully installed or removed.
After this operation, 0 B of additional disk space will be used.
Do you want to continue? [Y/n]
Setting up greyhole (0.9.61-1) ...
insserv: Service samba has to be enabled to start service greyhole
insserv: exiting now!
update-rc.d: error: insserv rejected the script header
dpkg: error processing package greyhole (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 greyhole
E: Sub-process /usr/bin/dpkg returned an error code (1)
root@debian:~# 
```

And here we can see the service masking:

```
root@debian:~# service samba start
Failed to start samba.service: Unit samba.service is masked.
root@debian:~# ls -l /lib/systemd/system/samba.service
lrwxrwxrwx 1 root root 9 Mar  7 06:10 /lib/systemd/system/samba.service -> /dev/null
```

More information about it here: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=769714

After this I tried all of this on Ubuntu 15.04 (since that's the first version with systemd), and everything happens exactly the same way as in Debian 8 above.

Incidentally, this appears to be an issue specific to Debian and its derivative distributions. I tried installing on CentOS 7 and it appeared to work without issue.

As a side issue, although I don't know if it's related to this, the greyhole deb places the file ```greyhole.systemd``` in ```/```, which seems like it's probably an error (both insofar as it should be called ```greyhole.service``` and it should be in ```/lib/systemd/system/``` if it's going to be installed, to the best of my understanding of systemd), but I don't think it's probably causing this issue since it falls back to the sysv init script anyway.

**Comment by tonymorella** [comment]
I can confirm all of the above.  Started with Ubuntu 15.04 and ran 
into the exact same issues. As a quick fix

nano /etc/init.d/samba:
Default-Start:     2 3 4 5
Default-Stop:      1

update-rc.d samba defaults
apt-get -f install

nano /etc/init.d/greyhole
Required-Start:    mysql 
Required-Stop:     mysql 

update-rc.d samba remove
updare-rc.d greyhole remove

Then set the start scripts up for systemd

**Comment by gboudreau** [comment]
I tried to install on a clean Ubuntu 15 install, and the error message I get is:

    status: Unable to connect to Upstart: Failed to connect to socket /com/ubuntu/upstart: Connection refused

Did you guys re-enable upstart..?

**Comment by gboudreau** [comment]
This should be fixed by a59a153a34ce529f385e6d5b5b62029882cd8081 and 402ab18fdafffed9a1480fd67cd55aa8a607e2c5.
Release 0.9.61-3 includes those.
Let me know if that fixes the issue for you.

**Comment by benmullins** [comment]
While I have no idea why your Ubuntu 15.04 install was still using Upstart (for my test with Ubuntu I just used a clean install of the 64-bit edition of Ubuntu Server 15.04, and the greyhole install was the first thing I did post-installation), but the 0.9.61-3 fixes seem to have resolved the issue. The install now completes correctly on fresh installs of Debian 8 and Ubuntu 15.04 for me.
